py/server/deephaven/table_factory.py

```diff
@@ -335,3 +340,88 @@ def ring_table(parent: Table, capacity: int, initialize: bool = True) -> Table:
         return Table(j_table=_JRingTableTools.of(parent.j_table, capacity, initialize))
     except Exception as e:
         raise DHError(e, "failed to create a ring table.") from e
+
+
+def function_generated_table(table_generator: Callable[[], Table],
+           source_tables: Union[Table, List[Table]] = None,
+           refresh_interval_ms: int = None,
+           exec_ctx: ExecutionContext = None) -> Table:
+    """
+    Creates an abstract table that is generated by running the table_generator() function. The function will first be
+    run to generate the table when this method is called, then subsequently either (a) whenever one of the
+    'source_tables' ticks or (b) after refresh_interval_ms have elapsed. Either 'refresh_interval_ms' or
+    'source_tables' must be set (but not both).
+
+    Function-generated tables can be used to produce dynamic tables from sources outside Deephaven. For example,
+    function-generated tables can create tables that are produced by arbitrary Python logic (including using Pandas or
+    numpy). They can also be used to retrieve data from external sources (such as files or websites).
+
+    The table definition must not change between invocations of the 'table_generator' function.
```

Comment:
```suggestion
    The table definition must not change between invocations of the 'table_generator' function, or an exception will be raised.
```
does this cause an exception?  If so, indicate in the docstring.